src/index/updater.rs

```diff
@@ -96,27 +98,46 @@ impl Updater {
       Some(progress_bar)
     };
 
-    let rx = Self::fetch_blocks_from(index, self.height, self.index_sats)?;
-
-    let (mut outpoint_sender, mut value_receiver) = Self::spawn_fetcher(index)?;
+    let rx = Self::fetch_blocks_from(self.index, self.height, self.index_sats)?;
 
     let mut uncommitted = 0;
-    let mut value_cache = HashMap::new();
     while let Ok(block) = rx.recv() {
-      self.index_block(
-        index,
-        &mut outpoint_sender,
-        &mut value_receiver,
-        &mut wtx,
-        block,
-        &mut value_cache,
-      )?;
+      if self.index.client.get_block_count()? - self.index.block_height()?.unwrap_or(Height(0)).n()
+        < 50
```

Comment:
This should probably be the maximum number of savepoints (=6). Anything higher is unnecessary unless you want to have extra reorg resistance during the initial indexing 😋